src/python_routing_v02/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -1208,77 +1280,102 @@ cpdef object compute_network_structured(
+                                         out_buf,
+                                         assume_short_ts)
+
+                    #Copy the output out
+                    for i in range(r.reach.mc_reach.num_segments):
+                        segment = get_mc_segment(r, i)
+                        flowveldepth[segment.id, timestep, 0] = out_buf[i, 0]
+                        flowveldepth[segment.id, timestep, 1] = out_buf[i, 1]
+                        flowveldepth[segment.id, timestep, 2] = out_buf[i, 2]
+
+            if gages_size:  # TODO: This loops over all gages for all reaches.
+                            # We should have a membership test at the reach loop level
+                            # so that we only enter this process for reaches where the
+                            # gage actually exists. We have the filter in place to
+                            # filter the gage list so that only relevant gages for a
+                            # particular network are present in the function call ---
+                            # adding the reach-based filter would be the next level.
+
+                for gage_i in range(gages_size):
+                    usgs_position_i = usgs_positions[gage_i]
+                    # TODO: It is possible to remove the following branching logic if
+                    # we just loop over the timesteps during DA and post-DA, if that
+                    # is a major performance optimization. On the flip side, it would
+                    # probably introduce unwanted code complexity.
+                    if (timestep < gage_maxtimestep and not isnan(usgs_values[gage_i,timestep-1])):
+                        flowveldepth[usgs_position_i, timestep, 0] = usgs_values[gage_i, timestep-1]
+                        # add/update lastobs_timestep
+                        lastobs_timestep[gage_i] = timestep - 1
+                        lastobs_values[gage_i] = usgs_values[gage_i, timestep-1]
+                    else:
+                        a = 120  # TODO: pull this a value from the config file somehow
+                        if lastobs_timestep[gage_i] < 0: # Initialized to -1
+                            da_decay_minutes = (timestep) * dt / 60 - time_since_lastobs_init[gage_i] # seconds to minutes
+                        else:
+                            da_decay_minutes = (timestep - lastobs_timestep[gage_i]) * dt / 60
 
-              elif r.type == compute_type.RESERVOIR_RFC:
-                run_rfc_c(r, upstream_flows, 0.0, 300, &reservoir_outflow, &reservoir_water_elevation)
-                flowveldepth[r.id, timestep, 0] = reservoir_outflow
-                flowveldepth[r.id, timestep, 1] = 0.0
-                flowveldepth[r.id, timestep, 2] = reservoir_water_elevation
+                        # replacement_value = f(lastobs_value, da_weight)  # TODO: we need to be able to export these values to compute the 'Nudge'
```

Comment:
I'm not sure if this has anything to do with the DA issues were currently experiencing, but printing flowveldepth[usgs_position_i, timestep, 0] before and after line 1374 shows two totally different results. I am not sure if the number is breaking here or not. When I compared this branch to a previous one where this step wasn't occurring at all, the results looked to be the same. Possibly influencing the error we are seeing in DA. 

                    printf("flowveldepth111: %d\t", flowveldepth[usgs_position_i, timestep, 0])
                    # replacement_value = f(lastobs_value, da_weight)  # TODO: we need to be able to export these values to compute the 'Nudge'
                    flowveldepth[usgs_position_i, timestep, 0] = simple_da_with_decay(lastobs_values[gage_i], flowveldepth[usgs_position_i, timestep, 0], da_decay_minutes, a)
                    printf("flowveldepth222: %d\t", flowveldepth[usgs_position_i, timestep, 0])